src/frequenz/sdk/actor/power_distributing/power_distributing.py

```diff
@@ -132,8 +132,32 @@ async def _run(self) -> None:  # pylint: disable=too-many-locals
         """
         await self._component_manager.start()
 
+        power_tasks: dict[str, asyncio.Task[None]] = {}
+
+        def cleanup(task: asyncio.Task[None]) -> None:
+            task_name = task.get_name()
+            assert task_name in power_tasks, "Task not found in power_tasks"
+
+            # Raise if there was an exception while processing the task
+            task.result()
+
+            del power_tasks[task_name]
+
         async for request in self._requests_receiver:
-            await self._component_manager.distribute_power(request)
+            task_name = f"power_request_ids={request.component_ids}"
+            if task_name in power_tasks:
+                result = Error(
+                    request=request,
+                    msg="A request for the same components is being processed. Skipped.",
+                )
+                await self._result_sender.send(result)
+                continue
+
+            power_tasks[task_name] = asyncio.create_task(
+                self._component_manager.distribute_power(request),
+                name=task_name,
+            )
+            power_tasks[task_name].add_done_callback(cleanup)
```

Comment:
This seems to actually be a perfect use case for a class I'm planning to add to `core` soon :tm: as part of the revamp of background service (I'm splitting out the task handling code to have a *TaskGroup* that is more useful for us for cases like this).

Reply:
Sounds great!

Reply:
* https://github.com/frequenz-floss/frequenz-core-python/pull/30